lib/App/Asciio/stripes/group.pm

```diff
@@ -38,8 +38,7 @@ for my $element (@{$elements})
 		my $text = $stripe->{TEXT} ;
 		
 		my $width = 0 ;
-		my $usc_length = usc_length($_) ;
-		map {$width  = $width < $usc_length ? $usc_length  : $width} split("\n", $text) ;
+		map {$width  = $width < usc_length($_) ? usc_length($_)  : $width} split("\n", $text) ;
```

Comment:
Good catch but you're computing the usc length twice and that an expensive operation.

blocks for map/grep, ... are normal blocks so you can declare variables in them, in this case to compute the usc_length just once.

Reply:
I know what you mean. I did calculate it twice here, but I verified that setting it to a variable didn't work, so I changed it back . This is the problem I encountered when I was working with group objects today . From the verification results, I have changed the right one, but the principle has to wait for me to learn before I can understand.

Reply:
here is an example:

my @l = map { my $x = $_ * 2 ; $x + $x + $x ; } (1, 2, 3) ;
print "@l\n" ;

as you can see $x is computed from $_ and then re-used multiple times.


also note that the use of map is a bit "not so good), map's result should be re-used, it's not a for loop as we use it in this code.

the code would be MUCH better using max()

something like:

$width = max( map { usc_length $_ } split("\n", $text) ;